flow/project.py

```diff
@@ -2247,7 +2352,12 @@ def _get_group_status(self, group_name, cached_status, ignore_errors=False):
                 job_op_id = group._generate_id(aggregate)
                 scheduler_status = cached_status.get(job_op_id, JobStatus.unknown)
                 completed = group._complete(aggregate)
-                eligible = not completed and group._eligible(aggregate)
+                eligible_ignore_conditions = (
+                    IgnoreConditions.POST if completed else IgnoreConditions.NONE
+                )
+                eligible = not completed and group._eligible(
+                    aggregate, ignore_conditions=eligible_ignore_conditions
+                )
```

Comment:
I'm confused about why this ignore conditions logic is necessary. The checks for `eligible` will short-circuit if ` completed is True`, so `group._eligible` is only called when `not completed`, in which case we should pass `IgnoreConditions.None`, which is the default. Am I missing something? If not, I think I saw the exact same logic in one of your new functions (probably the function getting aggregate/group status?).

Reply:
Here's the logic:
- `completed is False` ▶️ _some postcondition is not satisified_
- `completed is False` ▶️ we need to run the eligibility check.
- Without the `ignore_conditions`, the eligibility checks postconditions _again_ (if all preconditions are met).
- Since we already know that _some postcondition is not satisfied_, it is sufficient to determine eligibility only according to preconditions. This eliminates the need to check postconditions twice.

This will provide a benefit only when all preconditions are met (otherwise the eligibility check would exit at the first failed precondition).

Reply:
Ah I see, this is purely an optimization. I thought there was an issue with correctness. Please add a comment describing this logic (more briefly :)) in the code.